_posts/2019-04-10-aws-dashboard-console-using-arduino.md

```diff
@@ -0,0 +1,265 @@
+
+![Dashboard console](/img/aws-dashboard-console-using-arduino/dashboard-console.jpg "Dashboard console")
+
+## The AWS Dashboard Console Project
+
+The aim of this project was to create a prototype of the AWS dashboard console using Arduino compatible board and electronic components. The project was mainly a hobby project, but the idea behind was also to create something that I could use on my daily work.
+
+The console presents AWS alerts and codepipeline status by the LED lights and LCD screen. The console runs independently by polling AWS metrics over the WiFi. The console uses NodeMCU microcontroller with an integrated ESP8266 WiFi chip. Just plug-in the power and go. 
+
+![Dashboard architecture](/img/aws-dashboard-console-using-arduino/dashboard-architecture.png "Dashboard architecture")
+
+The architecture contains a NodeMCU board and AWS Lambda backend. The lambda serves AWS status information via the REST interface to the internet. The console polls the REST interface periodically and controls LED lights and LCD screen.
+
+### Arduino and NodeMCU/ESP8266 microcontroller
+
+[Arduino](https://www.arduino.cc) series boards, such as a popular Arduino Uno, contains a simple programmable microprocessor and set of GPIO connectors for the electronic components. Some of the board models has also an integrated WiFi chip which enables the internet connection. The Arduino compatible boards do not include an operating system which is the main difference compared to the popular Rasberry Pi series boards.
+
+[NodeMCU](http://www.nodemcu.com/index_en.html) is an Arduino compatible open source IoT board. It includes hardware which is based on the ESP8266/ESP-12 WiFi module. The NodeMCU is comparable to the common Arduino boards; it can be programmed by the Arduino IDE and it *stupidly* runs the code that is stored to its flash memory.
+
+### Circuit of the Console
+
+The circuit of the console contains following components:
+- 1 x NodeMCU board
+- 1 x LCD 2x16 characters I2C
+- 7 x LED lights
+- 7 x resistors for the LEDs
+- 1 x breadboard
+- wires
+
+![Dashboard circuit](/img/aws-dashboard-console-using-arduino/dashboard-circuit.png "Dashboard circuit")
+
+Basic layout of the circuit is a relative simple. The LEDs (and resistors) are connected to the NodeMCU IO pins. The LCD screen data wires are connected to the default I2C IO pins. The LCD screen power is connected to the NodeMCU 5V VIN pin. The NodeMCU board is powered by the USB connector. Components are assembled on the breadboard to minimize soldering.
```

Comment:
relatively simple

Ehkäpä: ..minimize soldering required.

Reply:
Basic layout of the circuit is a relative simple. --> The basic layout of the circuit is relatively simple.